mlfinlab/online_portfolio_selection/online_portfolio_selection.py

```diff
@@ -0,0 +1,296 @@
+
+    def conversion(self, _all_weights, _portfolio_return):
+        """
+        Converts the given np.array to pd.Dataframe for visibility
+
+        :param _all_weights: (np.array) portfolio weights for the time period
+        :param _portfolio_return: (np.array) cumulative portfolio returns for all periods
+        :return: (None) set all_weights and portfolio_return to pd.Dataframe
+        """
+        self.all_weights = pd.DataFrame(_all_weights, index=self.time, columns=self.asset_name)
+        self.portfolio_return = pd.DataFrame(_portfolio_return, index=self.time, columns=["Returns"])
+
+    def optimize(self,
+                 _optimize_array,
+                 _solver=cp.SCS):
+        """
+        Calculates weights that maximize returns over a given _optimize_array
+
+        :param _optimize_array: (np.array) relative returns of the assets for a given time period
+        :param _solver: (cp.SOLVER) set the solver to be a particular cvxpy solver
+        :return weights.value: (np.array) weights that maximize the returns for the given optimize_array
+        """
+
+        # initialize weights for optimization problem
+        weights = cp.Variable(self.number_of_assets)
+
+        # used cp.log and cp.sum to make the cost function a convex function
+        # multiplying continuous returns equates to summing over the log returns
+        portfolio_return = cp.sum(cp.log(_optimize_array * weights))
+
+        # Optimization objective and constraints
+        allocation_objective = cp.Maximize(portfolio_return)
+        allocation_constraints = [cp.sum(weights) == 1, cp.min(weights) >= 0]
+        # Define and solve the problem
+        problem = cp.Problem(objective=allocation_objective, constraints=allocation_constraints)
+        # if there is a specified solver use it
+        problem.solve(warm_start=True, solver=_solver)
+        return weights.value
+
+    def round_weights(self,
+                      threshold=1e-6):
+        """
+        Drops weights below a certain threshold
+
+        :param threshold: (float) drop all values below this threshold
+        :return (None): (np.array) sets all_weights as cleaned portfolio weights for the time period
+        """
+        # set all values below the threshold to 0
+        new_all_weights = np.where(self.all_weights < threshold, 0, self.all_weights)
+        # normalize the weights to sum of 1 in case the weights don't add up to 1
+        new_all_weights = np.apply_along_axis(lambda x: x / np.sum(x), 1, new_all_weights)
+        self.all_weights = new_all_weights
+
+    @staticmethod
+    def normalize(_weights):
+        """
+        Normalize sum of weights to one
+
+        :param _weights: (np.array) value of portfolio weights that has not been processed yet
+        :return (None): (np.array) normalize the input and sets self.weights
+        """
+        norm_weights = _weights / np.sum(_weights)
+        return norm_weights
+
+    def uniform_weight(self):
```

Comment:
All of your methods are public. Should most of them not rather be private? In python we make a method private by adding a underscore as such: `_method'